### PR TITLE
8274293: Build failure on macOS with Xcode 13.0 as vfork is deprecated

### PR DESCRIPTION
**Pull request by xpbob**
remove vfork() on Darwin

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8274293](https://bugs.openjdk.java.net/browse/JDK-8274293): Build failure on macOS with Xcode 13.0 as vfork is deprecated

### Reviewers
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**)
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5686/head:pull/5686` \
`$ git checkout pull/5686`

Update a local copy of the PR: \
`$ git checkout pull/5686` \
`$ git pull https://git.openjdk.java.net/jdk pull/5686/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5686`

View PR using the GUI difftool: \
`$ git pr show -t 5686`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5686.diff">https://git.openjdk.java.net/jdk/pull/5686.diff</a>

</details>

### CODE REVIEW

File: src/java.base/unix/native/libjava/ProcessImpl_md.c [hunk 1]
```diff
@@ -444,7 +444,8 @@ static int copystrings(char *buf, int offset, const char * const *arg) {
 __attribute_noinline__
 #endif
 
-/* vfork(2) is deprecated on Solaris */
+/* vfork(2) is deprecated on Solaris and Darwin */
```
Comment (AlanBateman) [inline src/java.base/unix/native/libjava/ProcessImpl_md.c:447]: Drive by comment is that we can probably drop "Solaris" from the comment.

**Review by AlanBateman — APPROVED** [review]
I think this looks okay but would be good for @RogerRiggs to also review.

**Review by tstuefe — COMMENTED** [review]
The JDK change is fine. AFAICS we have used posix_spawn by default on MacOS since at least 2015, and fork was not even an option on MacOS.

Wrt the hotspot portion, we now fall back to fork. I wonder what the footprint penalty is compared to vfork. `os::fork_and_exec` gets used mainly by error reporting (`-XX:OnError`=xxx), e.g. in OOM situations. We now may run into situations where error analysis tools are unable to spawn if the parent is too big. Maybe we should change `os::fork_and_exec()` to use posix_spawn instead?

**Review by tstuefe — APPROVED** [review]
.. I opened https://bugs.openjdk.java.net/browse/JDK-8274320 to track the proposed change of os::fork_and_exec() to posix_spawn. So I am fine with this change.

**Review by RogerRiggs — APPROVED** [review]
Looks good, vfork is not/was not being used on Mac OS.

**Pull request closed without merging**